test/unittests/state_transition_txcreate_test.cpp

```diff
@@ -1170,6 +1170,42 @@ TEST_F(state_transition, txcreate_from_blob_tx)
     expect.post[*tx.to].storage[0x01_bytes32] = 0x01_bytes32;
 }
 
+TEST_F(state_transition, txcreate_loop)
+{
+    rev = EVMC_OSAKA;
+    block.gas_limit = 30'000'000;
+    tx.gas_limit = block.gas_limit;
+    pre.get(tx.sender).balance = tx.gas_limit * tx.max_gas_price + tx.value + 1;
+
+    const auto deploy_container = eof_bytecode(bytecode(OP_INVALID));
+
+    const auto init_code = returncode(0, 0, 0);
+    const bytecode init_container = eof_bytecode(init_code, 2).container(deploy_container);
+
+    tx.type = Transaction::Type::initcodes;
+    tx.initcodes.push_back(init_container);
+
+    constexpr auto iterations = 500;
```

Comment:
as mentioned on the call, I'd also make a version with invalid EOF, and with a mix valid/invalid. But maybe enough to leave it for EEST.

There is also a twisted variant where the loop is handled by recursion, where an (erroneous) cache mechanism could clear the cache on every call depth level. Also possible better in EEST.